### PR TITLE
(#9) Dictionary Landing Page update

### LINKED ISSUE #9: Story: Dictionary Landing Page (URL is /)
## As a user, I would like to have a default (or “landing”) page when I first arrive at a Dictionary so that I can understand what the dictionary contains, and choose to either browse or search the dictionary. 

> **ESTIMATE** 3

### Acceptance Criteria

```gherkin
Scenario: User is able to access a default (or “landing”) page that displays all of the expected Dictionary elements for each dictionary. 
  Given user is on home page
  When user navigates to “Dictionary” tab
  Then user is on landing page for the selected Dictionary
    And page displays left navigation 
    And page title is “NCI Dictionary of Cancer Terms”
    And {} introductory text appears below the page title
    And search bar is displayed below introductory text
  Then “Starts with” and “Contains” options are displayed
    And “Starts with” is selected by default
  Then “Browse” option is displayed below search bar
    And browse option contains all alphabetic letters  
    And “#” appears at the end of the list 
    And each option appears as a link 
    And the page is showing the expand results for letter “A”
    And the URL does not include expand=A
```

```gherkin
Scenario: User is able to access a default (or “landing”) page that displays all of the expected Dictionary elements for each dictionary. 
  Given user is on home page
  When user navigates to “Dictionary” tab and clicks on Espanol
  Then user is on landing page for Diccionario
    And page displays left navigation 
    And page title is “Diccionario de cáncer”
    And {} introductory text appears below the page title
    Then “Empieza con” and “Contiene” options are displayed
    And “Empieza con” is selected by default
    And "keyword" search bar is displayed below introductory text and "Buscar" is displayed as the 
    submit button
  Then “Ojear” option is displayed below search bar
    And browse option contains all alphabetic letters  
    And “#” appears at the end of the list 
    And each option appears as a link 
    And the page is showing the expand results for letter “A”
    And the URL does not include expand=A
```

```gherkin
Scenario: Landing page metadata
  Given the user is viewing the dictionary landing page (/publications/dictionaries/cancer-terms)
  Then <meta name="robots" content="index" /> exists in the data for the page
```

```gherkin
Background
  Give language is set to "es"
Scenario: Spanish Landing page metadata
  Given the user is viewing the dictionary landing page (/espanol/publicaciones/diccionario)
  Then <meta name="robots" content="index" /> exists in the data for the page
```

### Resources:
* Feature: https://tracker.nci.nih.gov/browse/CGOVWCMS-30
* Web Pages:
    * https://www.cancer.gov/publications/dictionaries/cancer-terms
    * https://www.cancer.gov/publications/dictionaries/genetics-dictionary
    * https://www.cancer.gov/espanol/publicaciones/diccionario

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- /expand has already been built

### Tasks
- [x] Create basic search page with Search Box (no autosuggest), "Starts with" button, "Contains" button, and clickable A-Z list
- [ ] Make dictionary count available to insert into introductory text in Drupal
- [ ] Display intro text, with dictionary count, on the dictionary page
- [x] Set up default route to display expand=A (#10)
- [ ] Ensure page is set to indexable (robots = index) for search

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #9

### CODE REVIEW

File: support/src/setupProxy.js [hunk 1]
```diff
@@ -142,6 +143,11 @@ const middleware = (app) => {
     getTermTotalCount
   );
 
+  app.use(
+    '/api/Terms/count/:dictionary/:audience/:language',
+    getTermTotalCount
+  );
```
Comment (kate-mashkina) [inline support/src/setupProxy.js]: duplicated code?

Comment (seyilonge-nci) [reply support/src/setupProxy.js]: Thanks for the catch! Updated.

File: public/index.html [hunk 2]
```diff
@@ -64,7 +66,7 @@
           basePath: "",
           dictionaryEndpoint: "https://webapis-dev.cancer.gov/glossary/v1/",
           dictionaryIntroText:
-            "The NCI Dictionary of Genetics Terms contains technical definitions for more than 200 terms related to genetics.",
+            "<p>The NCI Dictionary of Genetics Terms contains technical definitions for more than {{term_count}} terms related to genetics. These definitions were developed by the <a title=\"Follow link\" class=\"external-link\" href=\"/publications/pdq/editorial-boards/genetics\" rel=\"nofollow\"> PDQ® Cancer Genetics Editorial Board</a> to support the evidence-based, peer-reviewed <a title=\"Follow link\" class=\"external-link\" href=\"/publications/pdq/information-summaries/genetics\" rel=\"nofollow\"> PDQ cancer genetics information summaries</a>.</p>",
```
[the remark this answers is not in the document]
Comment (seyilonge-nci) [reply public/index.html]: The endpoint does exist and returns a value in dev. My assumption was that it's the same with Cancer terms
`https://webapis-dev.cancer.gov/glossary/v1/Terms/count/Genetics/HealthProfessional/en`

**Comment by seyilonge-nci** [comment]
@VictoriaSunNIH @lburack
Please review PR https://react-app-dev.cancer.gov/glossary-app/pr-104/

**Comment by VictoriaSunNIH** [comment]
@seyilonge-nci , @lburack - Knowing this doesn't meet all the criteria (because we have to integrate with cancer.gov to see left nav, and language toggle), this passes for me on react-app URL.

**Comment by seyilonge-nci** [comment]
> @seyilonge-nci , @lburack - Knowing this doesn't meet all the criteria (because we have to integrate with cancer.gov to see left nav, and language toggle), this passes for me on react-app URL. Integration will be tested later.

👍

**Pull request merged**